fuel-core-interfaces/src/relayer.rs

```diff
@@ -0,0 +1,143 @@
+
+    /// Asumption is that validator state is already checked in eth side, and we can blidly apply
+    /// changed to db without checking if we have enought stake to reduce.
+    /// What needs to be done is to have validator set state and diff as saparate database values.
+    async fn insert_validator_set_diff(&mut self, eth_height: u64, stakes: &HashMap<Address, u64>) {
+        let _ = Storage::<u64,HashMap<Address,u64>>::insert(self, &eth_height,stakes);
+    }
+
+    async fn apply_current_validator_set(&mut self, changes: HashMap<Address,u64>) {
+        for (ref address,ref stake) in changes {
+            let _ = Storage::<Address,u64>::insert(self,address,stake);
+        }
+    }
+
+    /// get validator set for current eth height
+    async fn current_validator_set(&self) -> HashMap<Address,u64>;
+
+    /// set last finalized fuel block. In usual case this will be
+    async fn set_current_validator_set_eth_height(&self, block: u64);
+    /// Assume it is allways set as initialization of database.
+    async fn get_current_validator_set_eth_height(&self) -> u64;
+
+    /// get stakes difference between fuel blocks. Return vector of changed (some blocks are not going to have any change)
+    async fn get_validator_set_diff(
+            &self,
+            from_fuel_block: u64,
+            to_fuel_block: Option<u64>,
+    ) -> Vec<(u64,HashMap<Address, u64>)>;
+
+    /// current best block number
+    async fn get_block_height(&self) -> u64;
+
+
+    /// set newest finalized eth block
+    async fn set_eth_finalized_block(&self, block: u64);
+
+    /// assume it is allways set sa initialization of database.
+    async fn get_eth_finalized_block(&self) -> u64;
+
+    /// set last finalized fuel block. In usual case this will be
+    async fn set_fuel_finalized_block(&self, block: u64);
+    /// Assume it is allways set as initialization of database.
+    async fn get_fuel_finalized_block(&self) -> u64;
+}
+
+#[derive(Debug)]
+pub enum RelayerEvent {
+    //expand with https://docs.rs/tokio/0.2.12/tokio/sync/index.html#oneshot-channel
+    // so that we return list of validator to consensus.
+    GetValidatorSet {
+        /// represent validator set for current block and it is on relayer to calculate it with slider in mind.
+        eth_height: u64,
+        response_channel: oneshot::Sender<Result<HashMap<Address, u64>, RelayerError>>,
+    },
+    GetStatus {
+        response: oneshot::Sender<RelayerStatus>,
+    },
+    Stop,
+}
+
+pub use thiserror::Error;
+
+use crate::db::KvStoreError;
+
+#[derive(Error, Debug, PartialEq, Eq, Copy, Clone)]
+pub enum RelayerError {
+    #[error("Temp stopped")]
+    Stoped,
+    #[error("Temp ProviderError")]
+    ProviderError,
+    #[error("Validator Set not returned, waiting for eth client sync")]
+    ValidatorSetEthClientSyncing,
+    #[error("Asked for unknown eth block")]
+    InitialSyncAskedForUnknownBlock,
+}
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum EthSyncedState {
```

Comment:
"BaseChainSyncState" or "DaSyncState"

Reply:
DaSyncState seems nice. There is some inconsisntency with state, we dont use it per say inside relayer and those things will get flushed out when we know who needs it.